usr/share/rear/layout/prepare/GNU/Linux/110_include_lvm_code.sh

```diff
@@ -252,14 +252,29 @@ create_lvmvol() {
         warnraidline=""
     fi
 
-    cat >> "$LAYOUT_CODE" <<EOF
+    local fallbacklvopts
+    # Assume lvcreate had failed because of "Volume group ... has insufficient free space"
+    # which usually happens when the replacement disk is a bit smaller than the original disk
+    # so that the following fallback attempt to create a LV could work at least once per VG
+    # which is sufficient when the replacement disk is a bit smaller than the original disk
+    # because then the last LV that is created gets shrinked to the remaining space in the VG.
+    # In the lvopts string replace the exact size option of the form '-L 123456b'
+    # with an option to use all remaining free space in the VG via '-l 100%FREE'
+    # so e.g. 'lvcreate -L 123456b -n LV VG' becomes 'lvcreate -l 100%FREE -n LV VG'
+    fallbacklvopts="$( sed -e 's/-L [0-9b]*/-l 100%FREE/' <<< "$lvopts" )"
 
+    cat >> "$LAYOUT_CODE" <<EOF
 $ifline
-
-    LogPrint "Creating LVM volume '$vg/$lvname'; Warning: some properties may not be preserved..."
+    LogPrint "Creating LVM volume '$vg/$lvname' (some properties may not be preserved)"
     $warnraidline
-    lvm lvcreate $lvopts $vg <<<y
-
+    if ! lvm lvcreate $lvopts $vg <<<y ; then
+        LogPrintError "Failed to create LVM volume '$vg/$lvname' with lvcreate $lvopts $vg"
+        if lvm lvcreate $fallbacklvopts $vg <<<y ; then
+            LogPrint "Created LVM volume '$vg/$lvname' using fallback options lvcreate $fallbacklvopts $vg"
+        else
+            LogPrintError "Also failed to create LVM volume '$vg/$lvname' with lvcreate $fallbacklvopts $vg"
```

Comment:
I think I need to add a `false` here to let the whole 'if then else fi' command
exit with non zero exit state to let the diskrestore.sh script abort here
(because diskrestore.sh runs with 'set -e'), cf.
```
# if false ; then echo y ; else echo n ; fi && echo Y || echo N
n
Y

# ( set -e ; if false ; then echo y ; else echo n ; fi ; echo still there )
n
still there

# ( set -e ; if false ; then echo y ; else echo n ; false ; fi ; echo still there )
n
```

Reply:
Solved via
https://github.com/rear/rear/pull/2591/commits/e6e4edc92e963d0fc3169c2e4bdcbae42813400c